resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -81,6 +85,27 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
         configMapKey)
     }
 
+    val hadoopBootStrap = for {
+      hadoopConfigMap <- maybeHadoopConfigMap
```

Comment:
Never use `for..yield` with a single Option. Just use the other primitives. `for..yield` is primarily a shorthand for `Option#flatMap`.